File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const q = [];
   for(let i = 0; i < priorities.length; i++){
       q.push([priorities[i], i]);
    }
    const sorted = [...priorities].sort((a,b) => a-b);
    let idx = 0;
    let cnt = 0;
    console.log(sorted);
    while(idx !== q.length){
        const [fV,fI] = q[idx++];
        if(fV === sorted[sorted.length - 1]){
            sorted.pop();
            cnt++;
            if(fI === location) return cnt;
        }
        else{
            q.push([fV,fI]);
        }
    }
   
}
// 뭘 저장? [우선순위, 원래 인덱스] 쌍
// 언제 저장? 최대 우선순위가 아닐 때 큐 뒤에 다시 넣음
// 어디서 시작? 큐 앞에서부터 (인덱스 포인터 방식)
// 우선순위를 어떻게 관리하지?

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    q = deque([(p, i) for i, p in enumerate(priorities)])
    answer = 0

    while q:
        cur = q.popleft()
        if cur[0] < max(q, key=lambda x: x[0])[0]:
            q.append(cur)
        else:
            answer += 1
            if cur[1] == location:
                return answer